src/lib.rs

```diff
@@ -209,3 +209,58 @@ pub fn ixy_init(
         Ok(Box::new(device))
     }
 }
+
+impl IxyDevice for Box<dyn IxyDevice> {
+    fn init(pci_addr: &str, num_rx_queues: u16, num_tx_queues: u16) -> Result<Self, Box<dyn Error>> {
+        ixy_init(pci_addr, num_rx_queues, num_tx_queues)
```

Comment:
Isn't this just `unreachable!()`? Though I agree that `init` being part of the trait is odd. Perhaps we can remove it? (Would also be cleaner for the Virtio driver I'm working on as it only supports `num_rx_queues == num_tx_queues == 1`.)

Reply:
With this trait impl, `Box::<_>::init("...", 1, 1)` works just like a call to `::ixy_init`.

Reply:
The main problem that this addresses is that no specific impl of `IxyDevice` is available from the outside, and so one has to access `ixy_init` to use the library at all. So the `Self` in `Result<Self, _>` makes little sense since there can not be a concrete `Self` other than the `dyn IxyDevice` in the public interface.

Reply:
You're right, didn't think of that. Also somehow didn't notice before that the concrete impls aren't public which we may want to change. That's a separate issue though.

Reply:
Yes, the different driver `impl`s could be made public in the future. And yes, `init()` can be removed from the trait, 
@Bobo1239: Do you want to change these things before submitting your pull request?

Reply:
`ixy_init` would be very useful if more devices were auto-detected, even if the specific drivers were not public. So I don't think one change necessarily require the other.

It is not even strictly required for `Box<IxyDevice>` to implement `IxyDevice`, it just adds one extra level of indirection even when a concrete device is know. That is because without the impl on `Box`, the best generic way to store devices in a struct is to box them yourself or to require initialization other than `ixy_init` (which is impossible right now).

Note that removing `init` from the trait comes with the extra benefit of allowing an expanded impl to `Box<impl IxyDevice>` (compared to `Box<dyn IxyDevice>`). This would permit the user to choose between the concrete type in the box, and whether it need to be `Send` or any of the other auto-traits (not really a big issue right now, we are single threaded).

Reply:
Sure, the two changes do not depend on each other but I think it would be great to get both of them done. And removing `init()` and/or making the concrete driver implementations public doesn't mean that we have to remove `ixy_init()` as well, right? ;)

I'll merge this PR now, we can change those other things in future PRs. Sounds good?

Reply:
Yup, those are independent so go ahead.